metric/holder.go

```diff
@@ -0,0 +1,271 @@
+package metric
+
+import (
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const genTimeInterval = time.Second
```

Comment:
did you mean setTimeInterval?

Reply:
Now it's `updateTimeInterval` in xtime package